app/models/user.rb

```diff
@@ -5,6 +5,9 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable
   
   enum role: [:sender, :traveler]
-  has_many :journey
-  has_many :orders
+  
+  has_many :journeys
+
+  has_many :s_orders, foreign_key: :sender_id, class_name: 'Order'
+  has_many :t_orders, foreign_key: :traveler_id, class_name: 'Order'
```

Comment:
can you please explain this logic to me?

Reply:
Using multiple foreign keys in order table. traveler_id and sender_id and both are coming from user table. There was no space for self joins so i had to apply this technique. And it is working now. Basically its representing has_many orders